wgs_somatic-run-wrapper.py

```diff
@@ -0,0 +1,227 @@
+                    normal_samples.append(Sctx)
+                else:
+                    logger.info(f'Warning! {Sctx.slims_info["content_id"]} is not set as tumor or normal.')
+    
+    # make list of unique pair ids in current run
+    pair_ids_in_run = list(set(pair_ids_in_run))
+    logger.info(f'pair IDs in run: {pair_ids_in_run}')
+
+
+
+    # find tumor/normal pairs in the run and start pipeline
+
+    # get lists of ready samples and remove duplicates from list
+    tumor_samples_ready = list(set(get_samples_ready(tumor_samples, pair_ids_in_run, Rctx_run.run_tag)))
+    normal_samples_ready = list(set(get_samples_ready(normal_samples, pair_ids_in_run, Rctx_run.run_tag)))
+
+
+    # start the pipeline with the correct pairs. will use these arguments to start pipeline. some arguments are hardcoded right now, need to fix this. only considers barncancer hg38 (GMS-AL + GMS-BT samples) right now. could have outputdirs and arguments in config and get them from there. 
+    # this will only work for pairs. have to consider tumor only/normal only as well. 
+    # the arguments runtumor and runnormal could be "wrong" by doing it like this since they use run name of current run but if for example normal is in older run it has the wrong argument for "runnormal". this argument is not that important, it is only used to create a unique sample name. maybe it would be better discard/modify this argument than spending time on getting the correct value of it for samples from different runs. also, if fastqs come from more than one run, what will the value of this argument be then to be "correct"?...
+    # outputdir - need to consider if outputdir already exists (if sample has been run before and now it has new fastqs in current run, outputdir already exists). should old outputdir be moved to archive? 
+    for t in tumor_samples_ready:
+        if t.slims_info["content_id"] in started_samples:
+            continue
+        for n in normal_samples_ready:
+            if n.slims_info["content_id"] in started_samples:
+                continue
+            if t.slims_info['tumorNormalID'] == n.slims_info['tumorNormalID']:
+                logger.info(f'Starting wgs_somatic with arguments: \n runnormal: {Rctx_run.run_name} \n runtumor: {Rctx_run.run_name} \n tumorsample: {t.slims_info["content_id"]} \n normalsample: {n.slims_info["content_id"]} \n normalfastqs: {os.path.join(Rctx_run.run_path, "fastq")} \n tumorfastqs: {os.path.join(Rctx_run.run_path, "fastq")} \n outputdir: {os.path.join("/seqstore/webfolders/wgs/barncancer/hg38", t.slims_info["content_id"])} \n igvuser: barncancer_hg38 \n hg38ref: yes')
+                started_samples.extend((t.slims_info["content_id"], n.slims_info["content_id"]))
+
+
+   # next step here is to actually start the pipeline with these arguments
+
+
+
+if __name__ == '__main__':
+    try:
+        wrapper()
+    except KeyboardInterrupt:
+        pass
+    except Exception:
+        pass
```

Comment:
This should probably be handled somehow

Reply:
yep. I guess I could use (like in WOPR) traceback.format_exc() and send an email with the error